AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessor.swift

```diff
@@ -259,6 +267,19 @@ final class ItemListProcessor: StateProcessor<ItemListState, ItemListAction, Ite
         return []
     }
 
+    /// Subscribe to receive foreground notifications so that we can refresh the item list when the app is relaunched.
+    ///
+    private func setupForegroundNotification() {
+        services.notificationCenterService
```

Comment:
🤔 Why not a `for await _ in services.notificationCenterService.willEnterForegroundPublisher() {` like we do in the PM app?

Reply:
That's what I had originally, but I was running into lots of threading issues especially with the tests. We already have one `for await` (in streamItemList) that never terminates, so calling them both on `.appear` won't work. Calling it in a `Task` in `init` was causing the refresh to occur in other tests and other big threading problems. So it felt simpler to just use it directly as `AnyPublisher` rather than trying to map it into `for await`. Using the non-async version guarantees that we get the notification subscription set up (because it's synchronous) and we only see the single refresh that we expect in the tests.

Reply:
Interesting. Alright, makes sense.